src/render/PointArray.cpp

```diff
@@ -386,14 +386,85 @@ bool PointArray::loadFile(QString fileName, size_t maxPointCount)
     {
         g_logger.debug("Reordering field %d: %s", i, m_fields[i]);
         reorder(m_fields[i], inds.get(), m_npoints);
-        emit loadProgress(int(100*(i+1)/m_fields.size()));
+        emit loadProgress(int(100*(i+1)/(m_fields.size()+1))); // denominator +1 for permutation reorder below
     }
     m_P = (V3f*)m_fields[m_positionFieldIdx].as<float>();
 
+    // The index we want to store is the reverse permutation of the index above
+    // This is necessary if we want to mutate the data later
+    m_inds = std::unique_ptr<uint32_t[]>(new uint32_t[m_npoints]);
+    for (size_t i = 0; i < m_npoints; ++i) m_inds[inds[i]] = i; // Works for m_npoints < UINT32_MAX
+    emit loadProgress(int(100));
+
     return true;
 }
 
 
+void PointArray::mutate(std::shared_ptr<GeometryMutator> mutator)
+{
+    // Now we need to find the matching columns
+    auto npoints = mutator->pointCount();
+    const std::vector<GeomField>& mut_fields = mutator->fields();
+    auto mut_idx = mutator->index();
```

Comment:
I only just noticed - all the variable names in this function are using `underscore_case` which is a bit inconsistent.